### ISSUE TITLE
An apostrophe breaks syntax highlighting

### ISSUE THREAD

**Issue by dmnd** [orktes/atom-react#3]
Looks like the grammar thinks it's a Javascript string even when it's in the JSX part of the file:

![image](https://cloud.githubusercontent.com/assets/4427/3402387/7bd393de-fd5b-11e3-99d8-379b9692bf5c.png)

**Comment by rkovacevic** [comment]
![image](https://i.imgur.com/GNZRwkz.png)

Looks like there's still an issue. This is 0.12.10.

**Comment by reneklacan** [comment]
![screen shot 2015-12-10 at 22 27 27](https://cloud.githubusercontent.com/assets/1935686/11729003/4182d3a4-9f8d-11e5-83f9-2e3b358c4f21.png)

Still an issue on 1.2.4

**Comment by dmnd** [comment]
also, text inside JSX elements gets keyword highlighting:

![image](https://cloud.githubusercontent.com/assets/4427/11729069/2f764e7a-9f42-11e5-91d2-6a9bf53a502e.png)

**Comment by reneklacan** [comment]
@dmnd can you reopen this issue please?

**Comment by reneklacan** [comment]
:+1:

**Comment by dmnd** [comment]
I can't find how to reopen, so I just created #124.

**Comment by jesskrich** [comment]
This is still an issue. Is anyone working on the problem?
![screen shot 2016-04-20 at 1 32 21 pm](https://cloud.githubusercontent.com/assets/9681896/14684142/69cb5666-06fc-11e6-8503-a06bdf8936b4.png)

**Comment by dmnd** [comment]
It seems to regress often. Whoever fixes it should try to add a regression
test too.

**Comment by AgentME** [comment]
I've noticed that adding a `.` on the same line before the apostrophe is sometimes responsible for triggering this.

**Comment by pezza3434** [comment]
I also have this problem

**Comment by taygrave** [comment]
I have this problem right now too - despite an escape even
![screen shot 2016-05-23 at 17 30 19](https://cloud.githubusercontent.com/assets/6733241/15488529/1682e94a-210c-11e6-9699-9d1de80049c0.png)

**Comment by willhalling** [comment]
I am having this issue too so I've decided against using apostrophes. I just hope the grammar police don't come calling.

**Comment by joelwass** [comment]
Any update on this?

**Comment by syntacticsugar** [comment]
Running into this issue as well, I am coerced into foregoing apostrophes  :(  Any updates?

**Comment by mattp88** [comment]
The struggle is still real.  Any ideas?

**Comment by leanhdaovn** [comment]
Same issue in my project

**Comment by amit-coditas** [comment]
if i use this ’, instead of this ' it works. Not sure if it's grammatically right and allowed.
![image](https://cloud.githubusercontent.com/assets/19991098/17329145/ed8c576e-58de-11e6-84b0-5950f1c42271.png)

**Comment by jgnieuwhof** [comment]
Same issue here!

**Comment by Fakkio** [comment]
Same issue for me.
A temporary work around is to use &#39; instead of '.

**Comment by joshuadutton** [comment]
I'm having this issue too.

Workaround: put the text in a JavaScript var and insert it into JSX with the `{}` notation:
```javascript
  render() {
    const title = "Don't worry, be happy!";
    return (
      <div>
        <h1>{title}</h1>
      </div>
    );
  }
```

**Comment by salusinarduis** [comment]
This is really bothersome :(

**Comment by sendittokeith** [comment]
Yes, this is a sad limitation, makes a component almost unusable any time we need someone's stuff to have an apostrophe :-(

**Comment by andrewsunglaekim** [comment]
decorative apostrophe ftw

**Comment by julienbornstein** [comment]
Dirty workaround : use  `&apos; `entity 👎

**Comment by rchanou** [comment]
Yet another workaround:
```
<span>
  I{`'`}m a workaround. Let{"'"}s get some frosty chocolate milkshakes!
</span>
```

**Comment by sammy6464** [comment]
+ 1 for &#39;

**Comment by kengres** [comment]
I have the same problem, any updates?
![sans titre](https://cloud.githubusercontent.com/assets/14358992/20458251/da257fb6-aeb0-11e6-9284-7065a4c3ca03.png)

**Comment by justinzelinsky** [comment]
I'm having the same issue as @kengres.

**Comment by mattfysh** [comment]
Sames. Question mark also causes a problem - doesn't error-highlight subsequent text but stops syntax highlighting

**Comment by proftom** [comment]
same as @kengres

**Comment by srinath-ragh** [comment]
I have been working around the issue by adding a double apostrophe or double single quotes

**Comment by mattfysh** [comment]
Rest of team uses WebStorm and (understandably) don't want workarounds for atom-react in the code. Can we get word on if you guys have seen this issue and if its being prioritised? thanks

**Comment by kengres** [comment]
@proftom @srinath-ragh @justinzelinsky I've installed a package called [language-babel](https://atom.io/packages/language-babel) and the issue is gone...

**Comment by mattfysh** [comment]
related issue... https://github.com/orktes/atom-react/issues/212
Also looks like language-babel is no longer supported. Wish I had some free time to help with a fix!

**Comment by OriginalPenguin** [comment]
I'm hitting this bug as well.  It was first reported two and a half years ago!

Any work arounds?

**Comment by jedau** [comment]
I'm encountering this as well. I hoped this gets resolved soon. @OriginalPenguin's workaround is ingenious, so thanks for that!

**Comment by thuey** [comment]
This was a problem for me, too. Then I switched the language to JavaScript (JSX) and it started working. The only remaining problem is that syntax highlighting still breaks when the apostrophe isn't followed by another character. So something like dogs' would not work.

**Comment by thuey** [comment]
Actually, a co-worker discovered the language-babel plugin handles the trailing apostrophe as well.

**Comment by TuiKiken** [comment]
Still not working...

![screen shot 2017-01-10 at 01 23 48](https://cloud.githubusercontent.com/assets/959821/21786159/b696a8a4-d6d3-11e6-9521-519309b2d172.png)

**Comment by TheChuckster** [comment]
Issue is still around in 2017

**Comment by mattfysh** [comment]
Switch to language-babel ... much nicer

**Comment by wlingke** [comment]
@orktes Any updates to this?

**Comment by dovinmu** [comment]
language-babel is the way to go, thanks for the recommendation

**Comment by ryan-DESIGN** [comment]
How is this a thing?

**Comment by eldiyar7** [comment]
I would suggest installing language-babel package [https://atom.io/packages/language-babel]

**Comment by lenichols** [comment]
Why does this issue still exist...? just ran into it...

**Comment by andrewasfa** [comment]
Good, it's not just me. Was really annoying. Installing Language-babel solved the issue.

**Comment by benjaminu** [comment]
I believe this post on Stack Overflow should help. https://stackoverflow.com/q/32979512

```
...
return (
   <div>
     <p>{"I've seen the movie."}</p>
   </div>
 )
...
```

**Comment by vinagrito** [comment]
@orktes I guess you can close this one. As @kengres mentioned long ago there is a solution for this.
Installing `language-babel` on current 1.18.0 version (and assume that also on others) fixes the issue

**Comment by NathanCH** [comment]
Is there a reason to use this project in 2017? All features from atom-react are native in Atom except the code snippets.

**Comment by orktes** [comment]
@NathanCH Atom doesn't include proper indentation support (nested tags and attributes), highlighting and tag auto completion by default.

**Comment by faltysad** [comment]
language-babel solves this issue

**Comment by montanaflynn** [comment]
Our workaround is to use `&apos;`

**Comment by shuishoudage** [comment]
@kengres 's methods really work. don't waste time guys

**Comment by ashishytalele** [comment]
this code is usefull for me. you can use it as follow :
**<p>Let{"'"}s start by getting to know your business...</p>**

**Comment by ashishytalele** [comment]
this code is usefull for me. you can use it as follow :
**`<p>Let{"'"}s start by getting to know your business...</p>`**

**Comment by nsellers** [comment]
I installed language-babel and it resolved the issue for me.

**Comment by ZeroCool00** [comment]
use different class like Constant.js

declare your text there:

```
export const TEXT = 'Didn`t find what you want?';

app.js

import { TEXT } from '../Constants';

<Text>TEXT</Text>
```

**Comment by LukeBrandonFarrell** [comment]
Another workaround:
{'We won\'t do anything with you email. It\'s just to prevent spam.'}

A bit cleaner than the others suggested.

**Comment by lienne** [comment]
Just downloaded Atom yesterday and this is an issue for me as well :(

**Comment by benjaminu** [comment]
Follow @nseller's suggestion

https://github.com/orktes/atom-react/issues/3#issuecomment-371200678

**Comment by lienne** [comment]
Thanks! That worked wonders! :)

**Comment by krish-ryan** [comment]
Any updates on this?
Same issue here.

**Comment by ivanm376** [comment]
>language-babel

Additional **not breaking style** alternative: https://atom.io/packages/react

**Comment by StructByLightning** [comment]
nseller's comment worked for me but I had to disable atom-react.

**Comment by StructByLightning** [comment]
nseller's comment worked but I had to uninstall atom-react.

**Comment by nataliecardot** [comment]
language-babel worked, and I didn't have to uninstall atom-react OR restart Atom. Cool!

**Comment by alexanderconner** [comment]
I uninstalled atom-react and installed language-babel, and I'm still having this issue with both, even after uninstalling. 

![image](https://user-images.githubusercontent.com/24720260/46704660-fc00b480-cbf9-11e8-9572-29c71e4fa518.png)
